driver/manifest/manifest.go

```diff
@@ -211,13 +180,58 @@ func Load(path string) (*Manifest, error) {
 	if err != nil {
 		return m, err
 	}
+	if err := LoadBuildManifest(m, open); err != nil {
+		return m, err
+	}
 	m.Maintainers, err = Maintainers(open)
 	if err != nil {
 		return m, err
 	}
 	return m, nil
 }
 
+func extractVersion(s string) string {
+	if i := strings.LastIndexByte(s, ':'); i > 0 {
+		s = s[i+1:]
+	}
+	if i := strings.LastIndexByte(s, '-'); i > 0 {
+		s = s[:i]
+	}
+	return s
+}
+
+func LoadBuildManifest(m *Manifest, open OpenFunc) error {
+	f, err := open("build.yml")
```

Comment:
Not for this change, but I think we should revisit this interface with OpenFunc. There are several cases in this file where we have to do careful fiddling with filehandles to make sure everything gets closed properly, when in all the cases I can see we just want the whole file anyway.

I think I might come back and do a separate no-op simplification pass after this gets merged, since this change is actually solving a user-facing problem and should take precedence.

Reply:
I think it's pretty common to return read-closers from similar API methods. But for this case you are right, it always loads small text files that fit into memory just fine.

We are not in a hurry with this change, so I can make it here as a separate commit.

Reply:
Done